cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -189,16 +189,22 @@ handleSimpleNode
   -- otherwise the node won't actually start.
   -> IO ()
 handleSimpleNode p trace nodeTracers nc onKernel = do
+  meta <- mkLOMeta Notice Public
 
   let pInfo = Consensus.protocolInfo p
       tracer = toLogObject trace
 
   createTracers nc trace tracer
 
-  (publicIPv4SocketOrAddr
-    , publicIPv6SocketOrAddr
-    , localSocketOrPath) <- either throwIO return =<<
-                           runExceptT (gatherConfiguredSockets nc)
+  (publicIPv4SocketOrAddr, publicIPv6SocketOrAddr, localSocketOrPath) <- do
+    result <- runExceptT (gatherConfiguredSockets nc)
+    case result of
+      Right triplet -> return triplet
+      Left error -> do
+        traceNamedObject
+          (appendName "error" trace)
+          (meta, LogMessage (Text.pack (renderSocketConfigError error)))
+        throwIO error
```

Comment:
Also tracing the error because sometimes routing of `stdout` and `stderr` makes them hard to find.  The `throwIO` results in this:

```
$ journalctl -u cnode@1.service
eb 04 17:38:29 ubuntu-20 cnode[268589]: Listening on http://127.0.0.1:12791
Feb 04 17:38:29 ubuntu-20 cnode[268589]: Shutting down..
Feb 04 17:38:29 ubuntu-20 cnode[268589]: ClashingPublicIpv4SocketGiven
Feb 04 17:38:29 ubuntu-20 cnode[268589]: cardano-node: Configuration for the node's public IPv4 socket supplied both by config/cli and via systemd socket activation. Please use one or the other but not both.
Feb 04 17:38:29 ubuntu-20 systemd[1]: cnode@1.service: Main process exited, code=exited, status=1/FAILURE
Feb 04 17:38:29 ubuntu-20 systemd[1]: cnode@1.service: Failed with result 'exit-code'.
Feb 04 17:44:16 ubuntu-20 systemd[1]: Stopped Cardano Node 1.
```

The trace results in this:

```
$ cat node-1.json  | grep -i error
{"at":"2021-02-04T06:44:35.58Z","env":"1.25.1:12275","ns":["cardano.node.error"],"data":{"kind":"LogMessage","message":"Configuration for the node's public IPv4 socket supplied both by config/cli and via systemd socket activation. Please use one or the other but not both."},"app":[],"msg":"","pid":"270353","loc":null,"host":"ubuntu-2","sev":"Notice","thread":"5"}
```